diesel/src/pg/types/json.rs

```diff
@@ -0,0 +1,115 @@
+//! Support for JSON and `jsonb` values under PostgreSQL.
+
+extern crate serde_json;
+
+use std::io::prelude::*;
+use std::error::Error;
+
+use pg::Pg;
+use types::{self, ToSql, IsNull, FromSql};
+
+primitive_impls!(Json -> (serde_json::Value, pg: (114, 199)));
+primitive_impls!(Jsonb -> (serde_json::Value, pg: (3802, 3807)));
+
+impl FromSql<types::Json, Pg> for serde_json::Value {
+    fn from_sql(bytes: Option<&[u8]>) -> Result<Self, Box<Error+Send+Sync>> {
+        let bytes = not_none!(bytes);
+        serde_json::from_slice(&bytes)
+            .map_err(|e| Box::new(e) as Box<Error+Send+Sync>)
```

Comment:
Unfortunately, that does not seem to be the case. I tried removing it and got `expected trait std::error::Error, found enum pg::types::json::serde_json::Error`.